File: sdks/go/corthan/session.go
package corthan

import (
	"context"
	"fmt"
	"net/http"
)

// SessionService wraps Session operations.
type SessionService struct {
	t *TransportClient
}

// List retrieves active sessions.
func (s *SessionService) List(ctx context.Context) (*SessionListSuccess, error) {
	var result SessionListSuccess
	err := s.t.Request(ctx, http.MethodGet, "/sessions", nil, &result)
	if err != nil {
		return nil, err
	}
	return &result, nil
}

// Revoke terminates a session, invalidating its bound token.
func (s *SessionService) Revoke(ctx context.Context, id string) (*SessionRevokedSuccess, error) {
	path := fmt.Sprintf("/sessions/%s", id)
	var result SessionRevokedSuccess
	err := s.t.Request(ctx, http.MethodDelete, path, nil, &result)
	if err != nil {
		return nil, err
	}
	return &result, nil
}

File: sdks/node/src/services/session.ts
import { TransportClient } from "../transport.js";
import { SessionListSuccess, SessionRevokedSuccess } from "../types.js";

/**
 * Service for managing active sessions.
 */
export class SessionService {
  constructor(private transport: TransportClient) {}

  /**
   * Retrieves a list of active token sessions.
   */
  public async list(): Promise<SessionListSuccess> {
    return this.transport.request<SessionListSuccess>("GET", "/sessions");
  }

  /**
   * Invalidates a session by terminating its bound token.
   */
  public async revoke(id: string): Promise<SessionRevokedSuccess> {
    return this.transport.request<SessionRevokedSuccess>("DELETE", `/sessions/${id}`);
  }
}
